### ISSUE TITLE
PathCombine::Subtract missing a part of the result

### ISSUE THREAD

**Issue by andrewvarga** [Logicalshift/flo_curves#21]
So first thank you for this library, it seems to be really useful so far!

I have found one problem though and I'm not sure where to look next. I have the following setup of 2 paths: a grey rectangle and a "U" shaped path which is subtracted from it:
<img width="478" alt="Screenshot 2023-02-08 at 12 38 00" src="https://user-images.githubusercontent.com/911440/217519507-4d666acc-6f70-4d28-8c91-0b09842c8b1b.png">

What you see as grey pixels is what the result should be. However doing this: 
```rust
path_combine( PathCombine::Subtract(PathCombine::<SimpleBezierPath>::Path(rectangle, u_shape)))
```
produces this (the middle part is missing):
<img width="480" alt="Screenshot 2023-02-08 at 12 46 05" src="https://user-images.githubusercontent.com/911440/217521129-e9b89dbd-7998-469b-8e5c-401d9d236b9b.png">

I tried different accuracy settings, running `RemoveInteriorPoints` but didn't make any difference. Interestingly the other way around (subtracting the rectangle from the U shape works well as expected:
<img width="480" alt="Screenshot 2023-02-08 at 12 48 21" src="https://user-images.githubusercontent.com/911440/217521574-c26b6033-5b61-4d85-a84c-8b5ee8492513.png">

Do you have an idea what could be the problem? If it helps I can prepare a simplified example snippet too. Thank you!

**Comment by andrewvarga** [comment]
I think this has something to do with winding order: if the 2 paths have the same winding order (both clockwise or both counter clockwise) then the middle part is missing from the result, however if counter clockwise doesn't match (cw - ccw or ccw - cw) then it seems to work.
Is that the expected behaviour, and if so should I manually change the winding order of the paths in the subtraction?

**Comment by Logicalshift** [comment]
I think I've got this mostly fixed on the 0.7 branch. I've rewritten the exterior path finding algorithm as it was failing in this case.

It would have worked for any more complicated shape, but the simple shape here exposed a bug: if there was a path in the result where all the edges were moving away from every point, it wouldn't find a loop and thus not find a path.

Ie, if the result of merging the paths and applying the arithmetic operation looked like this:

![image](https://user-images.githubusercontent.com/4084632/218342243-232a00cf-e5ed-4713-8430-b2a447e6eff5.png)

No loop would be found for the center rectangle when the edges were in the direction shown (the algorithm would try all the directions at once and the shape wouldn't meet up at the start point). This is also why reversing the direction of the path works around the issue (adding an extra point in the middle of the rectangle would also work)

(Blue lines indicate external edges, and orange lines indicate hidden edges)

0.7 has a new algorithm that doesn't suffer from this issue, as well as some other issues such as not finding paths that only have one or two points. It fixes this problem but there are a few extra issues to fix before it's ready - three problematic test cases (rounding error generating one of those two-point paths that were previously ignored and a couple of test case that generates more than 64 edges on one point, which currently causes a failure)

**Comment by andrewvarga** [comment]
That's great, thank you for the clear explanation about what's going on behind the scenes!
Glad there is a solution, I don't need this very urgently, but it's very appreciated, so I'll be looking over it. Thanks!

**Comment by Logicalshift** [comment]
I've just released v0.7 of flo_curves, which has a fix for this issue (along with a few more issues I found along the way)

**Comment by andrewvarga** [comment]
Just wanted to confirm the fix worked well, thank you!
